Add deprecated toTraversable

ashawley opened the pull request:
The `toTraversable` method wasn't deprecated in 2.12, so it needs to be added back to fix clients who had used it in 2.12.  It should just be a proxy for `toIterable`, because `Traversable` is now just a type alias for `Iterable`.

After restoring `toTraversable` back, it needs to be marked deprecated in 2.13, so it can be removed in 2.14.

lrytz commented on src/library/scala/collection/Iterable.scala:
```diff
@@ -133,6 +135,13 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
+    *  @return An Iterable containing all elements of this $coll.
+    */
+  @deprecated("Use toIterable instead", "2.13.0")
+  def toTraversable: Traversable[A]
```
Why is this abstract? Could we define `@deprecated(..) final def toTraversable: Traversable[A] = toIterable` here?

ashawley replied:
That makes sense.  I was just mimicking the definition of `toIterator`.  I gave it a try and removed the abstract definition.  It actually makes this change a lot more minimal.

I don't think there's any reason it needs to be abstract.  Actually, it looks like overriding `toTraversable` was marked deprecated in 3cc99d7.  It wasn't possible to actually make it final in 2.12, because it was still being overridden in `ParIterableLike` and `TraversableProxyLike`, but we can try to do that here now.

julienrf replied:
I second @lrytz, just define it once and implement it at the same time.

ashawley committed:
Add deprecated toTraversable

The toTraversable method wasn't deprecated in 2.12, so it needs to be
added back to fix clients who had used it in 2.12.  It should just be
a proxy for toIterable, because Traversable is now just a type alias
for Iterable.  After adding toTraversable back, it needs to be marked
deprecated in 2.13, so it can be removed in 2.14.

This method used to be abstract, but there's no reason it still needs
to be going forward.  Overriding toTraversable was marked deprecated
in 3cc99d7.  It should have then been marked final in 2.12, but it
probably wasn't possible since toTraversable was still overridden in
ParIterableLike and TraversableProxyLike.

julienrf reviewed (commented):
That looks good! We could also add it to `StringOps` and `ArrayOps`, for the sake of completeness.

ashawley commented:
> We could also add it to `StringOps` and `ArrayOps`, for the sake of completeness.

Is there some `Predef` magic with implicits that make the tests I wrote for `Array` and `String` misleading?

ashawley commented:
I started looking in to whether it would be possible to add `toTraversable` to `ArrayOps` and `StringOps`.  It would be possible but it would be difficult since `toIterable` isn't in scope, yet.

I looked at 2.12, and it doesn't look like `toTraversable` was defined in `ArrayOps` or `StringOps`.  So people who may have disabled `Predef` would not have access `toTraversable`.

```
scalacOptions ++= Seq(
  "-Yno-predef",
  "-Yno-imports"
)
```

```scala
import scala.Array
import java.lang.String

// Error: value toTraversable is not a member of Array[T]
Array.empty.toTraversable 
// Error: value toTraversable is not a member of String
("": String).toTraversable 
```
https://scastie.scala-lang.org/jaFyka1jRh2EiS7N4HYhIA
